### PR TITLE
Fix the issue that the maximum width of the column does not change when a format is applied to the column

### PR DESCRIPTION
**Pull request by Kexi002**
When the getmaxwidth() function calculates the maximum width of a column, the format is not applied to the column. When the user sets the format, the maximum number of columns that the Dataset can display is still calculated according to the original data. However, the maximum number of displayed columns may change after setting the format.

Now the format of each column will be applied before calculating the maximum width of the column. The maximum number of columns that can be displayed will be changed correctly.

### CODE REVIEW

**Comment by sl-solution** [comment]
please add the `mapformats` keyword argument to `getmaxwidths`, `_show(io::IO, ::MIME"text/html", ds::AbstractDataset;...)`. Set its default value to `true`. Change your function in such a way that if `mapformats = true` then it calculates max widths based on the formatted values, otherwise, it must calculates max widths based on the actual values.

**Comment by sl-solution** [comment]
Any reason for putting everything inside the `if` condition? you could simply put `if` around the 
```
f = getformat(ds, col_idx)
col = f.(col)
```
part .

**Comment by Kexi002** [comment]
> Any reason for putting everything inside the `if` condition? you could simply put `if` around the
> 
> ```
> f = getformat(ds, col_idx)
> col = f.(col)
> ```
> 
> part .

If we just put another  FOR loop inside the `if `condition, we only need to determine `mapformat` once.

But if we put `if` around the that part:
```
for (......)
  if mapformat
    f = getformat(ds, col_idx)
    col = f.(col)
  end
end
```
We need to determine `mapformat` for every columns.

**Comment by Kexi002** [comment]
I just noticed that you do the same thing with show_eltype. I thought about this question, but I'm not sure which is more important: to simplify the code or improve a little bit efficiency (which is almost negligible).

File: src/abstractdataset/io.jl [hunk 1]
```diff
@@ -182,7 +194,7 @@ function _show(io::IO, ::MIME"text/html", ds::AbstractDataset;
     if get(io, :limit, false)
         tty_rows, tty_cols = displaysize(io)
         mxrow = min(mxrow, tty_rows)
-        maxwidths = getmaxwidths(ds, io, 1:mxrow, 0:-1, :X, nothing, true, buffer, 0) .+ 2
+        maxwidths = getmaxwidths(ds, io, 1:mxrow, 0:-1, :X, nothing, true, buffer, 0, mapformats) .+ 2
```
Comment (sl-solution) [inline src/abstractdataset/io.jl:197]: Refer to the previous comment: `mapformats` should be a keyword argument.

File: src/abstractdataset/io.jl [hunk 2]
```diff
@@ -44,18 +44,25 @@ function getmaxwidths(ds::AbstractDataset,
                       rowid::Union{Integer, Nothing},
                       show_eltype::Bool,
                       buffer::IOBuffer,
-                      truncstring::Int)
-    maxwidths = Vector{Int}(undef, size(ds, 2) + 1)
+                      truncstring::Int,
+                      mapformats::Bool)
```
Comment (sl-solution) [inline src/abstractdataset/io.jl:48]: Here, make `mapformats` as a keyword argument with default value of `true`. Update the docstring of the function accordingly.

**Review by sl-solution — CHANGES_REQUESTED** [review]

**Comment by sl-solution** [comment]
There is a conflict ("This branch has conflicts that must be resolved"), resolve it. Current version cannot be merged.

**Pull request merged**